src/util/eip712.ts

```diff
@@ -171,6 +172,9 @@ function getGjsonPrimitiveType(value: any) {
     if (typeof value === 'boolean') {
         return 'bool'
     }
+    if (Array.isArray(value) && value.length && typeof value[0] === 'string') {
```

Comment:
Might want to check that all values are string in this if statement (i.e. `value.every(item => typeof item === "string")`).